src/components/actorScriptButton.jsx

```diff
@@ -1,26 +1,22 @@
 import React from 'react';
+import { getCurrentActorCharacters } from './api';
 
 const ActorScriptButton = ({ actor, setCurrentActor, setCurrentCharacters, currentScript }) => {
   // on change, change the value of current actor
   const setActorHandler = async () => {
     setCurrentActor(actor);
-
-    // fetch request to get the current characters for the current actor
-    try {
-      const response = await fetch(`/api/script/${currentScript}/characters?actorId=${actor.id}`);
-
-      if (response.ok) {
-        setCurrentCharacters((await response.json()).map((c) => c.name));
-      } else {
-        console.error('Server Error:', response.body);
-      }
-    } catch (error) {
-        console.error(`error: ${error} when fetching current characters`);
+    // Get all the characters assigned to the current actor and update the currentCharacters list
```

Comment:
nit: Could we add a bit more of the "why" here? For example: `// We need to fetch the assigned characters of the actor we're switching to`

Reply:
Fixed!